src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,72 @@
+package mate.academy.service.impl;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+
+    private static final String HASHING_ALGORITHM = "SHA-256";
+    @Inject
+    private UserService userService;
```

Comment:
The `@Inject` annotation is used here, but the constructor also initializes `userService`. Ensure that only one method of dependency injection is used to avoid conflicts. Consider removing the constructor if the DI framework handles the injection.